packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,485 @@
+import type { IFocusTrapZoneProps } from '@fluentui/react/lib/FocusTrapZone';
+import type { FTZTestGlobals } from './types';
+
+/** Window with zero or more extra functions defined by some of the stories. */
+type FTZTestWindow = Cypress.AUTWindow & FTZTestGlobals;
+
 const ftzStoriesTitle = 'Components/FocusTrapZone/e2e';
 
+/**
+ * Calls `window.setProps()` -- this must be defined by the story being tested
+ */
+function setProps(props: IFocusTrapZoneProps) {
+  // Use should() to ensure that the call retries if setProps isn't defined yet
+  // (this can happen in headless mode, if the example isn't finished rendering)
+  cy.window().should(win => {
+    (win as FTZTestWindow).setProps!(props);
+  });
+}
+
 describe('FocusTrapZone', () => {
   before(() => {
     cy.visitStorybook({ qs: { e2e: '1' } });
   });
 
+  // These are basic tests of different props, but due to the reliance on focus behavior they're
+  // best done in the browser.
+  describe('Respects default and explicit prop values', () => {
+    beforeEach(() => {
+      cy.loadStory(ftzStoriesTitle, 'PropValues');
+    });
+
+    it('Focuses first child on mount', () => {
+      setProps({});
+
+      cy.focused().should('have.text', 'first');
+    });
+
+    it('Does not focus first child on mount with disableFirstFocus', () => {
+      setProps({ disableFirstFocus: true });
+
+      // Verify nothing is focused (note that document.activeElement will be body, but cy.focused()
+      // uses :focus, and that's not set on body even if it's active)
+      cy.focused().should('not.exist');
+    });
+
+    it('Can click children inside the FTZ', () => {
+      setProps({});
+
+      // wait for first focus to finish
+      cy.focused().should('have.text', 'first');
+
+      // focus inside the FTZ
+      cy.contains('mid').realClick();
+      cy.focused().should('have.text', 'mid');
+    });
+
+    it('Restores focus to FTZ when clicking outside FTZ', () => {
+      setProps({});
+
+      // wait for first focus to finish to avoid timing issue
+      cy.focused().should('have.text', 'first');
+
```

Comment:
You could maybe click mid or last before clicking outside so that you have a change from mid or last to first instead of having first twice in a row.

Reply:
For some reason that makes the rest of the test stop working, which may be due to function component conversion bugs. I'll leave it as-is for now but try that suggestion in the converted version.